### DISCUSSION TITLE
regression in Quarkus 2.8.0.Final when using Kover

### DISCUSSION THREAD

**Discussion by kenyee** [quarkusio/quarkus#24710] [Q&A] [answered]
when using Kover (https://github.com/Kotlin/kotlinx-kover), the dependency graph seems to have a circular dependency now in Gradle:
```
./gradlew check
* What went wrong:
Circular dependency between the following tasks:
:check
+--- :koverMergedReport
|    +--- :koverMergedHtmlReport
|    |    +--- :quarkusIntTest
|    |    |    \--- :check (*)
|    |    \--- :testNative
|    |         \--- :check (*)
|    \--- :koverMergedXmlReport
|         +--- :quarkusIntTest (*)
|         \--- :testNative (*)
+--- :koverMergedVerify
|    +--- :quarkusIntTest (*)
|    \--- :testNative (*)
\--- :koverVerify
     +--- :quarkusIntTest (*)
     \--- :testNative (*)

(*) - details omitted (listed previously)
```

Reverting to 2.7.5.Final fixes this.

Has anyone found a workaround?

[a comment by an automated account is not shown]

**Reply by glefloch** [reply]
Right, an issue as been open, I pushed a fix.

**Reply by pwlan** [reply]
Hi,

has the issue been fixed?

**Reply by gsmet** [reply]
@pwlan it's going to be fixed in 2.8.1.Final that I will release next week.

**Reply by pwlan** [reply]
Thanks for the info!

**Comment by kenyee** [comment, marked as answer]
Confirming this is fixed in 2.8.1

**Comment by kdubb** [comment]
@kenyee I'm curious how you are setting up Kover for Quarkus. Are you applying `quarkus-jacoco` for `@QuarkusTest` and `@QuarkusIntegrationTest` then using Kover to merge them?

Looking to convert because we almost exclusively use Kotlin and Jacoco is missing things.

**Reply by kenyee** [reply]
Kover uses jacoco underneath... When I set up kover, the quarkus jacoco extension wasn't available...

**Comment by kdubb** [comment]
The Kover Gradle plugin headline says it has the option to use the "Kover" engine; which is apparently another name for https://github.com/JetBrains/intellij-coverage.

I was interested in that engine only because it seems to support more Kotlin features.

**Reply by kenyee** [reply]
I can't find anything in the docs on how to configure it to use the intellij engine vs. jacoco but it claims to support both.

Ahh..found a bit of info in the migration docs for 0.7.0:
https://github.com/Kotlin/kotlinx-kover/blob/264e34b72c776fc47e8a11237bcc4a2b5a1de8e7/docs/migration-to-0.7.0.md

I've always used the defaults (never specified to use one or the other) which sounds like it's the Kover/intellij engine.

**Reply by kenyee** [reply]
Jacoco doesn't see a lot of the Kotlin hidden checks (like nullability checks) so you can never get to 100% coverage...I'm not convinced the Intellij engine will let you get there either but it's probably a bit better than Jacoco and might ignore the hidden Kotlin code.
